module/signature/aggregation.go

```diff
@@ -242,12 +261,12 @@ type PublicKeyAggregator struct {
 	// since the caller should not be aware of the internal non thread-safe algorithm.
 }
 
-// creates a new public key aggregator from all possible public keys
+// NewPublicKeyAggregator creates an index-based key aggregator, for the given list of authorized signers
 func NewPublicKeyAggregator(publicKeys []crypto.PublicKey) (*PublicKeyAggregator, error) {
 	// check for BLS keys
 	for i, key := range publicKeys {
 		if key == nil || key.Algorithm() != crypto.BLSBLS12381 {
-			return nil, engine.NewInvalidInputErrorf("key at index %d is not a BLS key", i)
+			return nil, fmt.Errorf("key at index %d is not a BLS key", i)
```

Comment:
Is this an exception because the caller is always ourselves who is providing the correct pub keys. 

Isn’t this also being used by the QC validator where the pub key might be a wrong one?

Reply:
In all cases, public keys (should) have undergone verification before they are committed into the protocol state:
* As part of staking, nodes (should) be submitting a 'proof of possession' (essentially a signature of some small piece of information, as evidence that they actually have the private key for the public one they are submitting).  As part of their verifying their 'proof of possession', we use their public key. If their public key was invalid, their staking transaction should already fail, in which they submit the public key + proof of possession. 
* For the random beacon private keys, those are generated during the DKG, where a majority of consensus nodes must consistently locally arrive at the same public key vector. I was assuming that the DKG protocol guarnatees that each public key share is valid (no-one might have a corresponding private key, if the respective participant failed the DKG ... nevertheless, it is a properly formatted public BLS key). 

At the point where we pull the public keys out of our storage, and find that these keys are wrong, the networks protocol state is already compromised. 


@tarakby is my assumption 👆 about DKG always producing valid keys correct? Would appreciate your thoughts on whether or not we could be encountering invalid BLS keys (not signatures) here during normal operation.